include/openthread/commissioner.h

```diff
@@ -114,17 +115,31 @@ typedef struct otCommissioningDataset
 
 #define OT_PSKD_MAX_SIZE 32 ///< Size of a Joiner PSKd (bytes)
 
+/**
+ * This enumeration defines a Joiner Info Typer.
+ *
+ */
+typedef enum otJoinerInfoType
```

Comment:
With C++11, should we be explicit with the type?

Reply:
Good idea. I agree we should use the explicit type in all core `enum` definitions.

In the case of public OT APIs/definitions, we still need to support `C` (not `C++`) so I am not sure if we can use the explicit type for these `enum` definitions.